src/topic_based_system.cpp

```diff
@@ -133,20 +133,34 @@ CallbackReturn TopicBasedSystem::on_init(const hardware_interface::HardwareInfo&
     }
     return default_value;
   };
+  // Load parameters from the hardware interface
+  time_from_start_ = std::stod(get_hardware_parameter("time_from_start", "0.1"));
+  position_cmd_only_ = get_hardware_parameter("position_cmd_only", "false") == "true";
+  joint_commands_type_ = get_hardware_parameter("joint_commands_type", "sensor_msgs/JointState");
+  block_joint_command_threshold_ = std::stod(get_hardware_parameter("block_joint_command_threshold", "0.5"));
+  trigger_joint_command_threshold_ = std::stod(get_hardware_parameter("trigger_joint_command_threshold", "1e-5"));
 
   // Add random ID to prevent warnings about multiple publishers within the same node
   rclcpp::NodeOptions options;
   options.arguments({ "--ros-args", "-r", "__node:=topic_based_ros2_control_" + info_.name });
 
   node_ = rclcpp::Node::make_shared("_", options);
 
-  if (auto it = info_.hardware_parameters.find("trigger_joint_command_threshold"); it != info_.hardware_parameters.end())
+  if (joint_commands_type_ == "sensor_msgs/JointState")
```

Comment:
Personally, I would make `joint_commands_type` a class enum and use switch statement for this part. It might have better performance and I think it looks cleaner. But you approach is totally fine tho. Up to u on whether to make the change.

Reply:
Overall good work!

Reply:
Make sense, but does this means using this parameters in sns_description xacro file need to be changed to
<param name="joint_commands_type"> [a int value] </param>
Does this lead extra misunderstanding？

Reply:
It is better keep sns_description param the same. One way u can do is to generate different enum of `joint_commands_type_` base on the param.